database/mongodb/mongodb.go

```diff
@@ -329,10 +329,11 @@ func (m *Mongo) ensureVersionTable() (err error) {
 // Utilizes advisory locking on the config.LockingCollection collection
 // This uses a unique index on the `locking_key` field.
 func (m *Mongo) Lock() error {
+	if !m.config.Locking.Enabled {
+		return nil
+	}
+
 	return database.CasRestoreOnErr(&m.isLocked, false, true, database.ErrLocked, func() error {
-		if !m.config.Locking.Enabled {
```

Comment:
I think we can always try to acquire the local lock before acquiring the db lock even if locking is disabled. IIRC, the locking enabled config was [added](https://github.com/golang-migrate/migrate/pull/448) for backwards compatibility and to avoid an extra network hop for existing clients

Reply:
ok, thanks. Done